channeldb/mp_payment.go

```diff
@@ -131,6 +131,37 @@ type MPPayment struct {
 	Status PaymentStatus
 }
 
+// TerminalInfo returns any HTLC settle info recorded, and any payment level
+// failure. Will be nil if no such info is recorded.
+func (m *MPPayment) TerminalInfo() (*HTLCSettleInfo, *FailureReason) {
+	for _, h := range m.HTLCs {
+		if h.Settle != nil {
+			return h.Settle, m.FailureReason
```

Comment:
is it possible to have a successful payment and a payment level failure?

Reply:
Yes! The settle will always override the `Failure` though, so we can consider just not returning the failure here.

Reply:
either is fine with me. we could also choose to return just `*lntypes.Preimage` instead of `HTLCSettleInfo` since the `SettleTime` is unused

Reply:
Will just keep returning `HTLCSettleInfo`, doesn't matter much I think